segmentstore/server/src/main/java/io/pravega/segmentstore/server/tables/EntryIteratorState.java

```diff
@@ -0,0 +1,137 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.server.tables;
+
+import com.google.common.base.Preconditions;
+import io.pravega.common.ObjectBuilder;
+import io.pravega.common.io.serialization.RevisionDataInput;
+import io.pravega.common.io.serialization.RevisionDataOutput;
+import io.pravega.common.io.serialization.VersionedSerializer;
+import io.pravega.common.util.ArrayView;
+import lombok.Getter;
+import lombok.SneakyThrows;
+
+import java.io.IOException;
+
+/**
+ * Represents the state of a resumable iterator.
+ */
+public class EntryIteratorState {
+    private static final Serializer SERIALIZER = new Serializer();
+
+    private static final int BOOLEAN_BYTES = 1;
+
+    @Getter
+    private final long position;
+
+    @Getter
+    private final boolean reachedEnd;
+
+    @Getter
+    private final boolean shouldClear;
+
+    @Getter
+    private final boolean deletionRecord;
+
+    /**
+     * Creates a new instance of the EntryIteratorState class.
+     *
+     * @param position The position of the TableEntry
+     * @param reachedEnd If the Entry is at the end of the segment (more recently appended).
+     * @param shouldClear Marks if the client should clear their state (provided start position has been truncated).
+     * @param deletionRecord The Entry read is marked for deletion.
+     */
+    EntryIteratorState(long position, boolean reachedEnd, boolean shouldClear, boolean deletionRecord) {
+        Preconditions.checkArgument(isValid(position), "Position must be at least 0 (a non-negative integer).");
+        this.position = position;
+        this.reachedEnd = reachedEnd;
+        this.shouldClear = shouldClear;
+        this.deletionRecord = deletionRecord;
+    }
+
+    boolean isValid(long position) {
+        return position >= 0;
+    }
+
+    @Override
+    public String toString() {
+        return String.format("Position = %s Reached End = %s Should Clear = %s", this.position, this.reachedEnd, this.shouldClear);
+    }
+
+    //region Serialization
+
+    /**
+     * Creates a new instance of the IteratorState class from the given array.
+     *
+     * @param data A byte array containing the serialization of an IteratorState. This must have been generated using
+     *             {@link #serialize()}.
+     * @return As new instance of the IteratorState class.
+     */
+    @SneakyThrows(IOException.class)
+    public static EntryIteratorState deserialize(byte[] data) {
+        return SERIALIZER.deserialize(data);
+    }
+
+    /**
+     * Serializes this IteratorState instance into an {@link ArrayView}.
+     *
+     * @return The {@link ArrayView} that was used for serialization.
+     */
+    @SneakyThrows(IOException.class)
+    public ArrayView serialize() {
+        return SERIALIZER.serialize(this);
+    }
+
+    private static class IteratorStateBuilder implements ObjectBuilder<EntryIteratorState> {
```

Comment:
You don't need to specify these fields in here. The `@Builder` annotation will take care of this for you. The only reason we need to declare this class is because we want it to implement `ObjectBuilder<>`.